### PR TITLE
Snap 892 new search by client

### PR DESCRIPTION
**Pull request by pokharelmanish**
### Jira Story

- Search by Client ID(https://osi-cwds.atlassian.net/browse/SNAP-892)

## Description
Able to search with the Client ID.

Note: This also includes Gustavo partial work. His PR looks good to me so I rebased and worked on my part. 
Client ID validation will come in the next PR. 

## Tests
- [x] I have included unit tests 
- [x] I have included feature tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

## Types of changes

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] I have ran all tests for the project.
- [ ] I have ran lint/rubocop check for the changed/new files.
- [ ] My code is in a stable state ready to be deployable, but not necessarily complete.
- [ ] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

**Comment by pokharelmanish** [comment]
Note: There is two code climate issue. We are looking into it.

File: app/javascript/sagas/fetchPeopleSearchSaga.js [hunk 1]
```diff
@@ -5,38 +5,81 @@ import {logEvent} from 'utils/analytics'
...
 
-const addressParams = (searchAddress) => {
+const personSearchParams = (personSearchFields) => {
   const params = {}
-  if (!searchAddress) { return {} }
+  if (!personSearchFields) { return {} }
 
-  if (searchAddress.county) {
-    params.county = searchAddress.county
+  if (personSearchFields.searchTerm) {
```
Comment (wmitchell-cambria) [inline app/javascript/sagas/fetchPeopleSearchSaga.js]: With so many fields, you may have better luck defining the object all at once, like:

```javascript
const params = {
  search_term: personSearchFields.searchTerm,
  last_name: personSearchFields.lastName,
  // ...
}

// If you need to, filter out falsy values afterwards:
Object.keys(params).forEach(key => params[key] || delete params[key])
```

Comment (pokharelmanish) [reply app/javascript/sagas/fetchPeopleSearchSaga.js]: Thanks @wmitchell-cambria

**Comment by pokharelmanish** [comment]
retest this please

File: app/helpers/query_builder_helper.rb [hunk 2]
```diff
@@ -11,6 +11,15 @@ module QueryBuilderHelper
   TRACK_SCORES = 'true'
   REQUIRE_FIELD_MATCH = 'false'
 
+  def formatted_date_of_birth(string)
+    return if string.nil?
+    date = DateTime.parse(string)
+    formatted_date = date.strftime('%m-%d-%Y')
+    formatted_date.to_s.downcase
+                  .gsub(%r{[-/]*(\d+)[-/]*}, '\1')
```
Comment (wmitchell-cambria) [inline app/helpers/query_builder_helper.rb]: If these regexes are exactly the same as the ones used below, perhaps we can factor them out into a helper method. That way, readers can understand what they do at a glance, without having to decipher the symbols.

Comment (pokharelmanish) [reply app/helpers/query_builder_helper.rb]: @ggomez-cambriasolutions has added this. I will let him decide on this.

**Pull request merged**